easybuild/easyblocks/m/matlab.py

```diff
@@ -141,24 +141,56 @@ def install_step(self):
             keys = keys.split(',')
 
         # Make one install for each key
-        for key in keys:
-            cmd = ' '.join([
-                self.cfg['preinstallopts'],
-                src,
-                '-v',
-                tmpdir,
-                '-inputFile',
-                self.configfile,
-                '-fileInstallationKey',
-                key,
-                self.cfg['installopts'],
-            ])
-            (out, _) = run_cmd(cmd, log_all=True, simple=False)
+        for i, key in enumerate(keys):
+
+            self.log.debug('Installing with key %s of %s', i + 1, len(keys))
+
+            if LooseVersion(self.version) >= LooseVersion('2020a'):
+                self.log.debug('Version is %s - using binary installer method', self.version)
+                try:
+                    with tempfile.NamedTemporaryFile() as fd:
```

Comment:
We might just as well move the whole section that creates self.configfile down into the installation step, and always regex in the file installation key into self.configfile.
That avoids creating a bunch of temp files.
But, with the file installation key written to a file, it must be protected from prying eyes and properly removed afterwards!

Reply:
@akesandgren Looking for a neat way to do this but in the case of versions < 2020a there's no requirement to regex the file installation key and the existing config file creation step is sufficient. (Earlier versions' installers take the `-fileInstallationKey` option.)
I could therefore keep the config file creation as is but make a regex for the _key_ line `^(# )?fileInstallationKey=.*` in the install step so that it can substitute and overwrite the installation key in place and avoid the temp file creation.